scripts/covid19indiaORG/main.py

```diff
@@ -0,0 +1,122 @@
+    """
+    # A valid response alway contains dates if there is timeseries data.
+    if 'dates' not in json:
+        return {}
+
+    timeseries_data = json['dates']
+
+    output = {}
+    # For every date, store the total data.
+    for date, data in timeseries_data.items():
+        if 'total' in data:
+            total = data['total']
+            output[date] = total
+    return output
+
+# A DataFrame containing ALL merged state data.
+table = pd.DataFrame({})
+
+# For every state in Config.STATE_APIS, query its data.
+for state in STATE_APIS:
+    # Every state has its own API.
+    api: str = state['API']
+    downloaded_data: Dict[str, Dict] = _download_data(api)
+
+    state_name: str = state['State']
+
+    # Example: WB for West Bengal.
+    state_abbrev: str = state['Abbreviation']
+
+    state_data = downloaded_data[state_abbrev]
+
+    # Get the timeseries for the given state abbreviation.
+    state_timeseries: Dict[str, Dict[str, int]] = \
+         _parse_timeseries(state_data)
+
+    # Store the data in a DataFrame, and transpose it.
+    # Transposing puts the dates in the Y-axis.
+    state_table = pd.DataFrame(state_timeseries).T
+
+    if state_name not in STATES:
+        continue
+
+    # Get the WikiDataId from the dictionary.
+    state_table['wikidataId'] = STATES[state_name]
+
+    # Store the DataFrame for this state under the main table.
+    table = pd.concat([table, state_table])
+
+    # The district data is stored under 'districts'.
+    if 'districts' not  in state_data:
+        continue
+    districts_data = state_data['districts']
+
+    # For every district in the state, parse the timeseries.
+    for district_name, data in districts_data.items():
+        district_timeseries = _parse_timeseries(data)
+        # Store the data in a DataFrame, and transpose it.
+        # Transposing puts the dates in the Y-axis.
+        district_table = pd.DataFrame(district_timeseries).T
+
+        if not (state_name in DISTRICTS and district_name in DISTRICTS[state_name]):
+            continue
+
+        # Get the WikidataId from the dictionary.
+        district_table['wikidataId'] = DISTRICTS[state_name][district_name]
+
+        # Store the DataFrame for this district under the main table.
+        table = pd.concat([table, district_table])
+
+# Rename the Y-axis to be "date".
+table.index.name = "date"
+
+# Calculate the active cases = confirmed cases - recovered cases.
+table['active'] = table['confirmed'] - table['recovered']
+
+# Get rid of any rows gthat don't have a wikidataId.
+table.dropna(subset=['wikidataId'], inplace=True)
+
+# Get rid of any rows that don't have at least one of the following.
+table.dropna(subset=['confirmed', 'deceased', 'tested', 'active', 'recovered'], thresh=1, inplace=True)
+
+# Only keep the following columns, the rest are not part of the import.
+table = table[['confirmed', 'deceased', 'tested', 'active', 'recovered', 'wikidataId']]
+
```

Comment:
Can you add an integration test for this main section? You'll probably want to put this whole chunk into a function called in main. Example integration test: https://github.com/datacommonsorg/data/blob/master/scripts/us_bea/states_gdp/import_data_test.py#L66

Reply:
Done